libs/knowledge-store/ragstack_knowledge_store/base.py

```diff
@@ -113,13 +113,19 @@ async def aadd_nodes(
         self,
         nodes: Iterable[Node],
         **kwargs: Any,
-    ) -> List[str]:
+    ) -> AsyncIterable[str]:
         """Add nodes to the knowledge store
 
         Args:
             nodes: the nodes to add.
         """
-        return await run_in_executor(None, self.add_nodes, nodes, **kwargs)
+        iterator = iter(await run_in_executor(None, self.add_nodes, nodes, **kwargs))
+        done = object()
+        while True:
+            doc = await run_in_executor(None, next, iterator, done)
```

Comment:
Should this use `aiter` and `anext`?

Reply:
`aiter` and `anext` work on `AsyncIterable`. Here `iterator` is an `Iterable`.